File: Skornel02.ETR.Common/Dtos/CourseDto.cs
using Skornel02.ETR.Common.Enums;

namespace Skornel02.ETR.Common.Dtos;

public record CourseDto(
    string CourseCode,
    string CourseSemester,
    int Hours,
    int Credits,
    string? CourseName,
    CourseType? CourseType,
    string ClassRoomAddress,
    string ClassRoomNumber,
    string ClassRoomName
);

File: Skornel02.ETR.Web/src/schemas/CourseDto.ts
import { z } from 'zod';
import { CourseType } from '../enums/coursetypes';

export const CourseDtoSchema = z.object({
	courseCode: z.string(),
	courseSemester: z.string(),
	hours: z.number(),
	credits: z.number(),
	courseName: z.string().optional(),
	courseType: z.nativeEnum(CourseType),
	classRoomAddress: z.string(),
	classRoomNumber: z.string(),
	classRoomName: z.string()
});

export type CourseDto = z.infer<typeof CourseDtoSchema>;
